packages/x-charts/src/context/HighlightedProvider/HighlightedProvider.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import useControlled from '@mui/utils/useControlled';
+import {
+  HighlightItemData,
+  HighlightedContext,
+  HighlightScope,
+  HighlightedState,
+  DeprecatedHighlightScope,
+} from './HighlightedContext';
+import { highlightedReducer } from './highlightedReducer';
+import { useSeries } from '../../hooks/useSeries';
+import { ChartSeriesType } from '../../models/seriesType/config';
+import { SeriesId } from '../../models/seriesType/common';
+
+export type HighlightedProviderProps = {
+  children: React.ReactNode;
+  /**
+   * The item currently highlighted. Turns highlighting into a controlled prop.
+   */
+  highlightedItem?: HighlightItemData;
+  /**
+   * The callback fired when the highlighted item changes.
+   *
+   * @param {HighlightItemData} highlightedItem  The newly highlighted item.
+   */
+  onHighlightChange?: (highlightedItem: HighlightItemData) => void;
+};
+
+const mergeDeprecatedOptions = (
+  options?: Partial<HighlightScope> | Partial<DeprecatedHighlightScope>,
+): HighlightScope => {
+  // @ts-expect-error deprecated behavior.
+  const { highlighted, faded, ...rest } = options ?? {};
+  return {
+    highlight: highlighted === 'series' ? 'same-series' : highlighted,
+    fade: faded === 'series' ? 'same-series' : faded,
```

Comment:
`other-series` initial idea was more on the `fade`, and was to fade all other series but the current one. Though it might not be necessary anymore?

I came up with it when creating `same-value|other-value`, since `value` would be too ambiguous. I think `series` is too ambiguous, are we applying it to the same series or different series?

Reply:
Since it's not needed to control highlights, would you be ok to revert 'same-series' to 'series' and do this breaking change in another PR dedicated to the renaming?

I would prefer to do this change in the PR for #12465 to be sure all namings aline.